### PR TITLE
[don't review] Test use of ringbuffer for shm notifications

### PR DESCRIPTION
**Pull request by pietern**
This is terrible code intended to just get a few measurements.

### CODE REVIEW

File: tensorpipe/transport/shm/socket.h [hunk 1]
```diff
@@ -210,13 +210,29 @@ class Socket final : public Fd, public std::enable_shared_from_this<Socket> {
   // Send file descriptor.
   template <typename... Fds>
   Error sendFds(const Fds&... fds) {
-    return sendFdsToSocket(fd_, fds...);
+    struct {
+    } nothing;
```
Comment (lerks) [inline tensorpipe/transport/shm/socket.h]: I was under the impression that the payload must contain at least one byte, but I cannot find any confirmation of that at the moment. Does that ring a bell? Do you know why it was one byte before?

File: tensorpipe/transport/shm/loop.cc [hunk 2]
```diff
@@ -222,6 +231,40 @@ void Loop::join() {
...
   thread_.join();
+  notificationMonitor_.join();
+}
+
+void Loop::notificationMonitor() {
+  while (!done_) {
+    uint32_t token;
+    auto ret = notification_->copy(token);
+    if (-ret == ENODATA) {
+      sched_yield();
```
Comment (lerks) [inline tensorpipe/transport/shm/loop.cc]: I was going to complain loudly about the busy waiting but this makes me kinda ok with it.

Comment (pietern) [reply tensorpipe/transport/shm/loop.cc]: We can still do better by adding a mutex and condition variable to the ring buffer header and having producers of these notifications signal there. Then this thread can be suspended if nothing is happening.

**Comment by pietern** [comment]
@lerks This is a total hack and we need a more disciplined approach :-)

**Comment by pietern** [comment]
This was a hacky proof of concept. The proper version is tested at #71.

**Pull request closed without merging**